examples/PoCTutorial/src/esdl_generator.py

```diff
@@ -0,0 +1,409 @@
+    source_1_inst_cost = 10e6   # eur
+    source_1_varop_cost = 100.0   # eur/MWh
+    source_1_fixop_cost = 6e5   # eur/MW
+    prod_1_coord = [52.00483, 4.37272]
+
+    # Source 2 parameters
+    source_2_power = 50e6       # W
+    source_2_inv_cost = 2e6     # eur/MW
+    source_2_inst_cost = 20e6   # eur
+    source_2_varop_cost = 200.0   # eur/MWh
+    source_2_fixop_cost = 9e5   # eur/MW
+    prod_2_coord = [51.99543, 4.36457]
+
+    # Joints
+    joint_1_coord = [52.00098, 4.36702]
+    joint_2_coord = [51.99596, 4.36951]
+    joint_1ret_coord = [52.00033, 4.36646]
+    joint_2ret_coord = [51.99537, 4.36895]
+
+    # Pipes
+    pipe_1_elbow_coord = [52.00353, 4.36593]
+    pipe_1ret_elbow_coord = [52.00363, 4.36492]
+    pipe_3_elbow_coord = [51.9898, 4.37191]
+    pipe_3ret_elbow_coord = [51.98922, 4.3717]
+
+    ######################################################
+    ##                  PROCESSING                      ##
+    ######################################################
+
+    esh = EnergySystemHandler()
+    energy_system = esh.create_empty_energy_system(name = 'Generated PoC tutoral')
+    main_area = energy_system.instance[0].area  # Logical high level area.
+
+    # Create heat carriers
+    heat_commodity_primary = esdl.HeatCommodity(id = str(uuid4()), name = 'HeatCommodityPrimary', supplyTemperature = temp_supply)
+    heat_commodity_ret = esdl.HeatCommodity(id = str(uuid4()), name = 'HeatCommodityRet', returnTemperature = temp_ret)
+    heat_carrs = esdl.Carriers(id = str(uuid4()))
+    heat_carrs.carrier.append(heat_commodity_primary)
+    heat_carrs.carrier.append(heat_commodity_ret)
+    for carrier in heat_carrs.carrier:
+        if carrier.name == 'HeatCommodityPrimary':
+            id_primary_carr = carrier.id
+        elif carrier.name == 'HeatCommodityRet':
+            id_ret_carr = carrier.id
+    esi = esh.energy_system.energySystemInformation
+    if not esi:
+        esi = esdl.EnergySystemInformation(id = str(uuid4()))
+        esh.energy_system.energySystemInformation = esi
+        esh.add_object(esi)
+    esi.carriers = heat_carrs
+
+    ##### Area 1 (demand 1)
+    area_1 = esdl.Area(id = str(uuid4()), name = 'Area_demand_1')
+    define_area(area_1, area_1_coord)
+    main_area.area.append(area_1)
+
+    heat_demand_1 = esdl.HeatingDemand(id = str(uuid4()), name = 'Heat_demand1')
+    area_1.asset.append(heat_demand_1)
+    add_geometry_point(heat_demand_1, demand_1_coord)
+    heat_demand_1.power = demand_1_power
+    
+    heat_demand_1.costInformation = esdl.CostInformation(id = str(uuid4()))  
+    costInf = heat_demand_1.costInformation
+    costInf.investmentCosts = create_cost_value_profile(demand_1_inv_cost, 'EURO', per_unit = 'WATT', per_multiplier = 'MEGA')
+    costInf.installationCosts = create_cost_value_profile(demand_1_inst_cost, 'EURO')
+    
+    in_carrier = esh.get_by_id_slow(id_primary_carr)
+    out_carrier = esh.get_by_id_slow(id_ret_carr)
```

Comment:
No need to get the carrier objects again every time. As long as you still have the same esdl, nothing reloaded or so, you can use the original objects you created, no need to search for their refrence again: heat_commodity_primary and heat_commodity_secondary.